main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"os"
+	"github.com/codegangsta/cli"
+	"fmt"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "fetch"
+	app.Usage = "download a file or folder from a specific release of a GitHub repo subject to the Semantic Versioning constraints you impose"
+	app.Version = getVersion(Version, VersionPrerelease)
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name: "repo",
+			Usage: "Required. Fully qualified URL of the GitHub repo.",
+		},
+		cli.StringFlag{
+			Name: "tag",
+			Usage: "The specific git tag to download, expressed with Version Constraint Operators.\n\tIf left blank, fetch will download the latest git tag.\n\tSee https://github.com/gruntwork-io/fetch#version-constraint-operators for examples.",
+		},
+		cli.StringFlag{
+			Name: "github-oauth-token",
+			Usage: "A GitHub Personal Access Token (https://help.github.com/articles/creating-an-access-token-for-command-line-use/)\n\tThis is used to authenticate fetch to private GitHub repos.",
+		},
+	}
+
+	app.Action = func(c *cli.Context) {
+
+		repoUrl := c.String("repo")
+		tagConstraint := c.String("tag")
+		githubToken := c.String("github-oauth-token")
+
+		// TODO: process repoFilePath and localFileDst args from command line
+		repoFilePath := "/"
+		localFileDst := "/Users/josh/temp"
+
+		// Validate required args
+		if repoUrl == "" {
```

Comment:
One nice thing about the github.com/jessevdk/go-flags library is it has built-in support for marking arguments as required or optional and providing default values. This reduces boilerplate and copy/paste (e.g. repeating the `--repo` arg below).